src/include/migraphx/op/nonmaxsuppression.hpp

```diff
@@ -308,6 +308,13 @@ struct nonmaxsuppression
         {
             return result;
         }
+
+        //#bound the max amount of boxes allowed to the max of boxes
+        if(max_output_boxes_per_class > args[1].get_shape().lens()[2])
+        {
+            max_output_boxes_per_class = args[1].get_shape().lens()[2];
+        }
```

Comment:
yea they are. Came up in resnext and thought it was odd since it renders part of the loop useless

![image](https://user-images.githubusercontent.com/107195283/214341540-f89ec17f-31bf-46c8-8a30-06f0c0349a3a.png)

Reply:
> it renders part of the loop useless
It doesn't. 

` while(not boxes_heap.empty() &&
                  selected_boxes_inside_class.size() < max_output_boxes_per_class)`. 

AND condition will  make sure we don't processes more than what we have.

Reply:
If the max is much larger than the amount of boxes that exist in the batch, select_boxes_inside_class.size() will never trigger.  That's why I added the change to bound this, otherwise in that case we'd need to process every value in the heap until it's empty() regardless of if we select every box.